node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/filter/AuthorizationFilter.java

```diff
@@ -84,13 +79,7 @@ private static void logAndReject(ErrorResponse response, ResponseHandler handler
 
     /** Read common name (CN) from certificate */
     private static String commonName(X509Certificate certificate) {
-        try {
-            X500Name subject = new JcaX509CertificateHolder(certificate).getSubject();
-            RDN cn = subject.getRDNs(BCStyle.CN)[0];
-            return IETFUtils.valueToString(cn.getFirst().getValue());
-        } catch (CertificateEncodingException e) {
-            throw new RuntimeException(e);
-        }
+        return X509CertificateUtils.getCommonNames(certificate).get(0);
```

Comment:
I think missing CN in certificate should be treated as a client error. Maybe just return `findFirst` from this method and `flatMap` it in the get certificate line?